hudi-client/hudi-client-common/src/test/java/org/apache/hudi/keygen/TestKeyGenUtils.java

```diff
@@ -124,7 +124,6 @@ public void testExtractRecordKeys() {
     String[] s3 = KeyGenUtils.extractRecordKeys("id:1,id2:__null__,id3:__empty__");
     Assertions.assertArrayEquals(new String[] {"1", null, ""}, s3);
 
-    // keys with ':' are not supported
```

Comment:
Forgot to remove it in https://github.com/apache/hudi/pull/12120 after review based changes in expected behavior.